packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -136,10 +136,10 @@ class AnnotationResult<T> {
 ///  * [RenderView.compositeFrame], which implements this recomposition protocol
 ///    for painting [RenderObject] trees on the display.
 abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
-  final List<VoidCallback> _callbacks = <VoidCallback>[];
+  final Map<int, VoidCallback> _callbacks = <int, VoidCallback>{};
 
   void _fireCompositionCallbacks(bool includeChildren) {
-    for (final VoidCallback callback in List<VoidCallback>.of(_callbacks)) {
+    for (final VoidCallback callback in List<VoidCallback>.of(_callbacks.values)) {
```

Comment:
I think .toList is the fast one since the exact type of the receiver is known in the implementation

Reply:
toList calls of

Reply:
🤦‍♂️

Reply:
https://github.com/dart-lang/sdk/blob/main/sdk/lib/_internal/vm/lib/array.dart#L169

Looks like if you give List.of an iterable that isn't backed by a list it first converts it to a list. so calling toList should be universally better

Reply:
https://github.com/dart-lang/sdk/blob/8aeeffb84825fa91c8b270afe2cdf77fad1cf1d5/sdk/lib/core/iterable.dart#L469-L471

Reply:
RIP